p2p/net/swarm/swarm_dial.go

```diff
@@ -303,6 +303,8 @@ func (s *Swarm) dial(ctx context.Context, p peer.ID) (*Conn, error) {
 	ila, _ := s.InterfaceListenAddresses()
 	remoteAddrs = addrutil.Subtract(remoteAddrs, ila)
 	remoteAddrs = addrutil.Subtract(remoteAddrs, s.peers.Addrs(s.local))
+	remoteAddrs = s.filterAddrs(remoteAddrs)
```

Comment:
we may want to special case the error here, where we may have addrs, but if we filter them all out, state that as a different error. basically:

```go
remoteAddrs = addrutil.Subtract(remoteAddrs, ila)
remoteAddrs = addrutil.Subtract(remoteAddrs, s.peers.Addrs(s.local))
if len(remoteAddrs) == 0 {
  err := errors.New("peer has no addresses")
  logdial["error"] = err
  return nil, err
}

remoteAddrs = s.filterAddrs(remoteAddrs)
if len(remoteAddrs) == 0 {
  err := errors.New("all peer addresses filtered out")
  logdial["error"] = err
  return nil, err
}
```